### DISCUSSION TITLE
Watermeter negative drift

### DISCUSSION THREAD

**Discussion by Zenyte-zz** [jomjol/AI-on-the-edge-device#2594] [Q&A]
Hallo Community, 
ich muss leider auf deutsch schreiben da mein englisch nicht wirklich ausreicht um mein Problem zu beschrieben.

Folgendes, meine Wasseruhr driftet langsam in einen negativen wert d.h. laienhaft gesprochen fließt mein Wasser wieder zurück in die Versorgungsleitung, wir sprechen hier von ca. 0,5 - 1 Liter über einen gewissen Zeitraum. Keine Ahnung ob das normal ist.

Siehe Logfile, Raw reduziert sich und daraus resultiert dann der Fehler Neg.Rate und ich erhalte in meiner Hausautomation keinen Wert mehr.

An welchem Stellrad muss ich in der Config drehen das er diesen minimalen negativ drift nicht als fehler sieht?
"Allow Negative Rates" will ich ungern ausschalten das es ja vorkommen kann das tatsächlich ein falscher wert 
ermittelt wird. 

`[0d09h17m13s] 2023-09-01T05:51:39	<ERR>	[POSTPROC] main: Raw: 129142.7, Value: , Status: Neg. Rate - Read:  - Raw: 129142.7 - Pre: 129143.6 
[0d09h19m13s] 2023-09-01T05:53:39	<ERR>	[POSTPROC] main: Raw: 129142.6, Value: , Status: Neg. Rate - Read:  - Raw: 129142.6 - Pre: 129143.6 
[0d09h21m13s] 2023-09-01T05:55:39	<ERR>	[POSTPROC] main: Raw: 129142.6, Value: , Status: Neg. Rate - Read:  - Raw: 129142.6 - Pre: 129143.6 
[0d09h23m13s] 2023-09-01T05:57:39	<ERR>	[POSTPROC] main: Raw: 129142.5, Value: , Status: Neg. Rate - Read:  - Raw: 129142.5 - Pre: 129143.6 
[0d09h25m13s] 2023-09-01T05:59:39	<ERR>	[POSTPROC] main: Raw: 129142.5, Value: , Status: Neg. Rate - Read:  - Raw: 129142.5 - Pre: 129143.6 
[0d09h27m13s] 2023-09-01T06:01:39	<ERR>	[POSTPROC] main: Raw: 129142.4, Value: , Status: Neg. Rate - Read:  - Raw: 129142.4 - Pre: 129143.6 
[0d09h29m13s] 2023-09-01T06:03:39	<ERR>	[POSTPROC] main: Raw: 129142.3, Value: , Status: Neg. Rate - Read:  - Raw: 129142.3 - Pre: 129143.6 
[0d09h31m13s] 2023-09-01T06:05:39	<ERR>	[POSTPROC] main: Raw: 129142.3, Value: , Status: Neg. Rate - Read:  - Raw: 129142.3 - Pre: 129143.6 
[0d09h33m13s] 2023-09-01T06:07:39	<ERR>	[POSTPROC] main: Raw: 129142.2, Value: , Status: Neg. Rate - Read:  - Raw: 129142.2 - Pre: 129143.6 
[0d09h35m13s] 2023-09-01T06:09:39	<ERR>	[POSTPROC] main: Raw: 129142.1, Value: , Status: Neg. Rate - Read:  - Raw: 129142.1 - Pre: 129143.6 
[0d09h37m13s] 2023-09-01T06:11:39	<ERR>	[POSTPROC] main: Raw: 129142.1, Value: , Status: Neg. Rate - Read:  - Raw: 129142.1 - Pre: 129143.6 `

PS: es hat mich Stunden gekostet bis ich bemerkt habe das "Maximum Rate Value" in abhängigkeit der "Decimal Shift" steht, da ich den Wert in Liter auslesen möchte habe ich das komma um 3 Stellen verschoben und deshalb ist bei mir Maximum Rate Value auf 350 und nicht wie empfohlen 0,05 ich hoffe ich habe es richtig verstanden denn erst seit dieser änderung erhalte ich keinen Fehler mehr "Rate too high"

**Comment by Zenyte-zz** [comment]
Meine Config sieht so aus, Wasseruhr ist Analog

`[TakeImage]
;RawImagesLocation = /log/source
WaitBeforeTakingPicture = 5
;RawImagesRetention = 15
Demo = false
Brightness = 0
Contrast = 0
Saturation = 0
LEDIntensity = 5
ImageQuality = 12
ImageSize = VGA
FixedExposure = false

[Alignment]
InitialRotate = 0.0
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
AlignmentAlgo = default
FlipImageSize = false
/config/ref0.jpg 226 103
/config/ref1.jpg 330 146

[Digits]
Model = /config/dig-cont_0611_s3_q.tflite
CNNGoodThreshold = 0.5
;ROIImagesLocation = /log/digit
;ROIImagesRetention = 3
main.dig1 243 134 28 50 false
main.dig2 272 134 28 50 false
main.dig3 301 134 28 50 false

[Analog]
Model = /config/ana-class100_0170_s1_q.tflite
;ROIImagesLocation = /log/analog
;ROIImagesRetention = 3
main.ana1 374 202 75 75 false
main.ana2 339 287 75 75 false
main.ana3 252 322 75 75 false
main.ana4 161 283 75 75 false

[PostProcessing]
main.DecimalShift = 3
main.AnalogDigitalTransitionStart = 7
PreValueUse = true
PreValueAgeStartup = 720
main.AllowNegativeRates = false
main.MaxRateValue = 350
main.MaxRateType = AbsoluteChange
main.ExtendedResolution = false
main.IgnoreLeadingNaN = false
ErrorMessage = true
CheckDigitIncreaseConsistency = false

;[MQTT]
;Uri = mqtt://IP-ADRESS:1883
;MainTopic = watermeter
;ClientID = watermeter
;user = USERNAME
;password = PASSWORD
RetainMessages = false
HomeassistantDiscovery = false
;MeterType = other

;[InfluxDB]
;Uri = undefined
;Database = undefined
;user = undefined
;password = undefined
;main.Measurement = undefined
;main.Field = 

;[InfluxDBv2]
;Uri = undefined
;Database = undefined
;Org = undefined
;Token = undefined
;main.Measurement = undefined
main.Field = undefined

;[GPIO]
;IO0 = input disabled 10 false false 
;IO1 = input disabled 10 false false 
;IO3 = input disabled 10 false false 
;IO4 = built-in-led disabled 10 false false 
;IO12 = input-pullup disabled 10 false false 
;IO13 = input-pullup disabled 10 false false 
LEDType = WS2812
LEDNumbers = 2
LEDColor = 150 150 150

[AutoTimer]
AutoStart = true
Interval = 2

[DataLogging]
DataLogActive = true
DataFilesRetention = 3

[Debug]
LogLevel = 1
LogfilesRetention = 3

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = pool.ntp.org
;Hostname = undefined
RSSIThreshold = -75
CPUFrequency = 160
SetupMode = false`

**Comment by leifnel** [comment]
Wenn der Wasserzähler rückwärts läuft, müssen Sie sich an Ihren Wasserversorger/Zählerbesitzer wenden.
Es muss ein Rückschlagventil vorhanden sein, das verhindert, dass (Schmutz-)Wasser aus Ihrer Anlage in das System zurückfließt und das Wasser anderer Personen verunreinigt.
Zumindest in Dänemark liegt es in der Verantwortung des Wasserversorgers, wahrscheinlich auch bei Ihnen. (Google translate)

If the water meter is running backwards, you will need to contact your water supplier/meter owner.
There must be a check valve to prevent (dirty) water from your system from flowing back into the system and contaminating other people's water.
At least in Denmark it is the responsibility of the water supplier, probably yours too.

**Comment by leifnel** [comment]
I translated the original message:
Hello community,
Unfortunately, I have to write in German because my English isn't really sufficient to describe my problem.

The following, my water meter is slowly drifting into a negative value, i.e. in layman's terms my water flows back into the supply line, we are talking about approx. 0.5 - 1 liter over a certain period of time. I don't know if that's normal.

See log file, Raw is reduced and this results in the Neg.Rate error and I no longer receive any value in my home automation.

Which dial do I have to turn in the config so that it doesn't see this minimal negative drift as an error?
I don't want to turn off "Allow Negative Rates" because it can actually happen that the value is incorrect
is determined.
---
PS: it took me hours until I noticed that "Maximum Rate Value" depends on the "Decimal Shift". Since I want to read the value in liters, I moved the decimal point by 3 places and that's why it's Maximum Rate Value for me to 350 and not 0.05 as recommended I hope I understood it correctly because only since this change did I no longer get the error "Rate too high"
